packages/jellyfish-api-core/__tests__/category/masternode/updatemasternode.test.ts

```diff
@@ -0,0 +1,576 @@
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { AddressType } from '../../../src/category/wallet'
+import { RpcApiError } from '@defichain/jellyfish-api-core'
+import { Testing, TestingGroup } from '@defichain/jellyfish-testing'
+import { RegTestFoundationKeys } from '@defichain/jellyfish-network'
+
+describe('Update Masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForWalletCoinbaseMaturity()
+
+    // enable updating
+    await client.masternode.setGov({
+      ATTRIBUTES: {
+        'v0/params/feature/mn-setowneraddress': 'true',
+        'v0/params/feature/mn-setoperatoraddress': 'true',
+        'v0/params/feature/mn-setrewardaddress': 'true'
+      }
+    })
+    await container.generate(1)
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should updateMasternode ownerAddress with bech32 address', async () => {
+    const initialAddress = await client.wallet.getNewAddress()
+    const masternodeId = await client.masternode.createMasternode(initialAddress)
+
+    await container.generate(20) // wait for masternode to be enabled
+
+    const masternodesBefore = await client.masternode.listMasternodes()
+    const masternodesLengthBefore = Object.keys(masternodesBefore).length
+
+    const newAddress = await client.wallet.getNewAddress()
+    await client.masternode.updateMasternode(masternodeId, {
+      ownerAddress: newAddress
+    })
+
+    // wait for masternode to be enabled
+    await container.generate(20)
+
+    // Test update masternode again during TRANSFERRING
+    const expectTransferringMN = await client.masternode.getMasternode(masternodeId)
+    expect(expectTransferringMN[masternodeId].state).toStrictEqual('TRANSFERRING')
+
+    const anotherNewAddress = await client.wallet.getNewAddress()
+    const promise = client.masternode.updateMasternode(masternodeId, {
+      ownerAddress: anotherNewAddress
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow(`Masternode ${masternodeId} is not in 'ENABLED' state`)
+
+    await container.generate(45)
+
+    const masternodesAfter = await client.masternode.listMasternodes()
+    const masternodesLengthAfter = Object.keys(masternodesAfter).length
+
+    expect(masternodesLengthAfter).toStrictEqual(masternodesLengthBefore)
+    expect(masternodesAfter[masternodeId]).toBeTruthy()
+    expect(masternodesAfter[masternodeId]).toStrictEqual({
+      ownerAuthAddress: newAddress,
+      operatorAuthAddress: initialAddress,
+      rewardAddress: '',
+      creationHeight: expect.any(Number),
+      resignHeight: expect.any(Number),
+      resignTx: expect.any(String),
+      collateralTx: expect.any(String),
+      state: expect.any(String),
```

Comment:
Can we use a deterministic state here?

Reply:
Added tests. https://github.com/JellyfishSDK/jellyfish/pull/1847/commits/f4620aa0953105f0f1ec2b19fab856a3c10c8b71

Reply:
I think it can be even more deterministic @imkven.

Reply:
Added test: https://github.com/JellyfishSDK/jellyfish/pull/1847/commits/399b6131c37ff2acf33f96380f85d51e4623c4b2